### ISSUE TITLE
Text icon, title justification

### ISSUE THREAD

**Issue by RobLeachman** [Snarling/bitburner-scripts#9]
I thought to go ahead and create an issue for the enhancement we're discussing on Discord to make the title bar even better. Please:

- Add a new default class textIcon that just includes the required style, or just make this is default style for .icon span (the code below), or whatever makes sense
- Explain (ELI5) how we can use new SVG's for alternate icons, like https://www.svgrepo.com/collection/business-finance-bold-icons/
- Help me fix the vertical justification of the box title.  I need to learn more CSS here is how it sits right now: https://imgur.com/a/HywKzZG

It was excellent to find the fix for #1 in the discord this morning, thanks!

```
import {createBox} from "box/box";
/** @param {NS} ns */
export async function main(ns) {
    let iconFromCharacter=char=>`<span style="font: 25px 'Lucida Console';line-height: 0.9;vertical-align:top;display:inline-block;width:25px;text-align:center">${char}</span>`;

    createBox("Test $ as icon","Lorem ipsum dolor sit amet",iconFromCharacter("$"))
    createBox("Test % as icon","Lorem ipsum dolor sit amet",iconFromCharacter("%"))
}
```

**Comment by Snarling** [comment]
Should all be addressed in last update. For SVGs it was previously a bit more work than I thought, but sizing should be taken care of now with the updated css. Unfortunately different SVGs need to be colored in different ways, so that part still needs to be done manually. Monochrome SVGs should use only transparent and currentColor as the colors, usually one as the background and one as a fill or stroke color.

**Comment by RobLeachman** [comment]
Awesome! Is the default plaintext icon style working as intended? [The font is different than the title text](https://imgur.com/a/B9aHg0F)

Would you please show me how to use an alternate SVG, I'm lost about injecting it and setting fill/stroke?

The bugfixes are of course perfect. Nice work and thanks!

**Comment by RobLeachman** [comment]
These all work for me, paste from my code today
`let content = "The tortoise lays on its back";`

Default SVG icon
`    //let box = doc.querySelector(".box._bladeburner") || createBox("Bladeburner", "", "\ueb0e", "_bladeburner");`

Default style plaintext
`    //let box = doc.querySelector(".box._bladeburner") || createBox("Bladeburner", "", "<span>B</span>", "_bladeburner");`

Fancy plaintext
`    const iconFromCharacter = char => '<span style="font: 25px 'Lucida Console';line-height: 0.9;vertical-align:top;display:inline-block;width:25px;text-align:center">B</span>';`
    `let box = doc.querySelector(".box._bladeburner") || createBox("Bladeburner", "", iconFromCharacter(), "_bladeburner");`

`    box.body.innerHTML = content; //Replaces the body content whether it's an old or new item`

**Comment by Snarling** [comment]
For SVGs, here's the process I use:

1. Find SVG. For an example I'll use this cat SVG icon found [here](https://www.veryicon.com/icons/object/material_design_icons/cat-42.html?p=3).
2. Open the SVG file in your browser.
3. CTRL-u to view the source. It should just be an SVG element with some other stuff inside it.
4. From here, I usually paste it into an SVG editor like Boxy SVG. From here I resize it so it fits within the canvas and there's an acceptable margin on all sides. Holding shift while resizing will maintain the aspect ratio.
5. From here I bring up the elements panel.
6. If there's just one path, I'll test out adding fill="red" to see if that does what I want. In this case it does, so I'll change it to fill=currentColor.
7. On the main SVG element, right click and copy outer SVG.
8. I usually delete any empty parts of the SVG, in this case there is a \<defs\> element with nothing in it. Usually everything but viewbox can be removed from the main svg element. 
9. Use the SVG directly as an icon. Here's the SVG and the result:

```
import {createBox} from "box/box"
export async function main(ns) {
  let svg=`<svg viewBox="0 0 500 500"><path d="M 244.565 144.756 L 213.163 146.88 C 192.858 122.798 135.957 62.119 79.293 62.119 C 79.293 62.119 32.78 132.006 78.348 225.267 C 65.362 244.863 57.335 255.016 55.682 278.39 L 10.114 285.237 L 15.072 308.375 L 56.627 302.237 L 59.932 319 L 22.864 341.194 L 33.961 362.207 L 68.196 341.194 C 95.348 398.803 164.054 428.08 244.565 428.08 C 325.076 428.08 393.782 398.803 420.934 341.194 L 455.169 362.207 L 466.266 341.194 L 429.198 319 L 432.503 302.237 L 474.058 308.375 L 479.016 285.237 L 433.448 278.39 C 431.795 255.016 423.768 244.863 410.782 225.267 C 456.35 132.006 409.837 62.119 409.837 62.119 C 353.173 62.119 296.272 122.798 275.967 146.88 L 244.565 144.756 M 173.734 215.587 C 186.72 215.587 197.344 226.211 197.344 239.197 C 197.344 252.183 186.72 262.807 173.734 262.807 C 160.748 262.807 150.124 252.183 150.124 239.197 C 150.124 226.211 160.748 215.587 173.734 215.587 M 315.396 215.587 C 328.382 215.587 339.006 226.211 339.006 239.197 C 339.006 252.183 328.382 262.807 315.396 262.807 C 302.41 262.807 291.786 252.183 291.786 239.197 C 291.786 226.211 302.41 215.587 315.396 215.587 M 220.955 286.418 L 268.175 286.418 L 251.648 319.236 C 256.37 334.347 269.592 345.444 285.883 345.444 C 305.48 345.444 321.299 329.625 321.299 310.028 L 333.104 310.028 C 333.104 335.999 311.855 357.249 285.883 357.249 C 268.175 357.249 252.829 347.568 244.565 333.638 C 236.301 347.568 220.955 357.249 203.247 357.249 C 177.276 357.249 156.026 335.999 156.026 310.028 L 167.831 310.028 C 167.831 329.625 183.65 345.444 203.247 345.444 C 219.538 345.444 232.76 334.347 237.482 319.236 L 220.955 286.418 Z" fill="currentColor"></path></svg>`
  createBox("Test cat icon","The icon is a cat! ".repeat(20), svg)
}
```
![image](https://user-images.githubusercontent.com/84951833/164271373-024a4ac5-3dbc-4d2a-a5df-0f181357c8a3.png)

**Comment by RobLeachman** [comment]
I did it! The razor blade icon of my dreams haha. Thanks much!
![image](https://user-images.githubusercontent.com/5489496/164576187-9e930a06-876e-44de-aef8-f3faf26813d4.png)
